Paper: Network visualizations with Pyvis and VisJS

boludo00 opened the pull request:
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.

Commit:
first draft scipy submission

Commit:
fix broken citation

sandhya212 commented:
Pyvis is a Python module that enables visualizing and interactively manipulating network graphs in the Jupyter notebook, or as a standalone web application TO quickly explore graph visualizations and investigate data relationships. Pyvis is also customizable.

Visualization is a key component to analyzing networks. The problem becomes more evident when handling massive datasets. Pyvis is a step in the right direction and the preliminary results presented here look promising.

Comments:
1. The fact that attributes such as node color, size, and layout would need to be manually entered is kind of crippling. Especially when it comes to scalability.

2. Is there an upper limit to number of nodes Pyvis can handle without significantly slowing down.

3. For the GoT experiment, please mention the number of characters (or nodes) in the data. You can shine light on a specific pair of characters known to be frequent and show that Pyvis does successfully depict that interaction.

4. Is the input to the GoT experiment, an adjacency matrix between characters?

5. Please number figures and place captions for ease of readability

summerswallow-whi committed:
Fix code link

Commit:
make changes based on feedback

Commit:
Merge branch '59_gperrone' of github.com:WestHealth/scipy_proceedings into 59_gperrone

Commit:
add bib

deniederhut commented:
Hey @boludo00 ! Thanks to some awesome behind the scenes work by @stargaser, we now have experimental support for publishing papers with ORCIDs. If you have one and would like to add it to your paper, it's as simple as adding an `:orcid:` tag under your name in the paper header. [You can see an example of how to do this in the `00_vanderwalt` example paper](https://raw.githubusercontent.com/scipy-conference/scipy_proceedings/2020/papers/00_vanderwalt/00_vanderwalt.rst). If you don't have one, no worries! You can still publish your paper as is.

boludo00 commented:
1. General practice for node customization in Pyvis consists of using the
`add_nodes` method (or `add_node` with looping mechanism) with list
operations in order to batch process your attributes.
2. Pyvis is built on top of VisJS, so performance is really limited by
VisJS implementation and client hardware.
3. Added in last commit
4. Yes, updated paper to mention this.
5. Figures have been added and labeled.

On Thu, Jun 18, 2020 at 10:17 AM Sandhya Prabhakaran <
notifications@github.com> wrote:

> Pyvis is a Python module that enables visualizing and interactively
> manipulating network graphs in the Jupyter notebook, or as a standalone web
> application TO quickly explore graph visualizations and investigate data
> relationships. Pyvis is also customizable.
>
> Visualization is a key component to analyzing networks. The problem
> becomes more evident when handling massive datasets. Pyvis is a step in the
> right direction and the preliminary results presented here look promising.
>
> Comments:
>
>    1.
>
>    The fact that attributes such as node color, size, and layout would
>    need to be manually entered is kind of crippling. Especially when it comes
>    to scalability.
>    2.
>
>    Is there an upper limit to number of nodes Pyvis can handle without
>    significantly slowing down.
>    3.
>
>    For the GoT experiment, please mention the number of characters (or
>    nodes) in the data. You can shine light on a specific pair of characters
>    known to be frequent and show that Pyvis does successfully depict that
>    interaction.
>    4.
>
>    Is the input to the GoT experiment, an adjacency matrix between
>    characters?
>    5.
>
>    Please number figures and place captions for ease of readability
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipy-conference/scipy_proceedings/pull/556#issuecomment-646190737>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACACLIOU4TOIFVHOMCZHXG3RXJD4LANCNFSM4NOHLK2Q>
> .
>

deniederhut commented:
Hi @sandhya212 ! Have the recent changes to the paper addressed your concerns?

sandhya212 commented:
Thank you authors for addressing my suggestions. They look well. 

To fully round up your paper, you can also add a 'Future directions' section (or add this in the conclusion) where one point you could discuss are your ideas for scalability (regarding my comment #2).

deniederhut commented:
Hi @boludo00 ! Have you had time to think about a future directions section to your paper?

Commit:
add a future directions section in conclusion

boludo00 commented:
@deniederhut Hi, yes! I just added a short section in my conclusion to wrap this up.

deniederhut commented:
Hi everyone! Friendly reminder that the open review period ends 2 July -- the committee will be making final decisions on each paper this Friday.

deniederhut commented:
Hi @sandhya212 ! Are you happy with the recent updates to the submission?

cbcunc commented on papers/giancarlo_perrone/ourbib.bib:
```diff
@@ -0,0 +1,44 @@
+  url={https://visjs.github.io/vis-network/docs/network/physics.html},
+  year= {2020},
+}
+ @misc{ barneshut,
```
Please provide a proper citation for Barnes-Hut other than Wikipedia. There is even one in the Wikipedia article: J. Barnes & P. Hut (December 1986). "A hierarchical O(N log N) force-calculation algorithm". Nature. 324 (4): 446–449. Bibcode:1986Natur.324..446B. doi:10.1038/324446a0

cbcunc replied:
Thank you for the changes. However, the full citation is not rendering (see http://procbuild.scipy.org/download/WestHealth-59_gperrone), I think because some of the field values at enclosed in double quotes instead of bibtex curly braces?

boludo00 replied:
Full citation for Barnes Hut appears in latest commit.

cbcunc replied:
Still not getting the DOI number provided for Barnes:  doi:10.1038/324446a0

boludo00 replied:
sorry, done now

cbcunc commented on papers/giancarlo_perrone/ourbib.bib:
```diff
@@ -0,0 +1,44 @@
+  title = {Network of Thrones},
+  url = {https://www.macalester.edu/~abeverid/thrones.html}
+}
+@InProceedings{networkx,
```
Please provide the DOI for SciPy Proceedings: doi:10.25080/issn.2575-9752

cbcunc reviewed (changes requested):
Minor updates to citations requested as part of final proceedings committee walk-through.

Commit:
reference barnes hit source + doi

Commit:
Fix Bibtex Citation

Commit:
no quotations in bib for barnes

Commit:
Merge branch '59_gperrone' of github.com:WestHealth/scipy_proceedings into 59_gperrone

Commit:
try reformat booktitle

Commit:
restructure bib for barnes

Commit:
doi for barnes

Commit:
miss comma

cbcunc reviewed (approved):
Thanks you for your updates!
